lib/modules/manager/kubernetes/extract.ts

```diff
@@ -37,14 +37,22 @@ export function extractPackageFile(
   return deps.length ? { deps } : null;
 }
 
+// Comes from https://github.com/distribution/reference/blob/v0.6.0/regexp.go
+// Extracted & converted with https://go.dev/play/p/lyNTyNMvIai
+const imageRegex =
+  /((?:(?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))*|\[(?:[a-fA-F0-9:]+)\])(?::[0-9]+)?\/)?[a-z0-9]+(?:(?:[._]|__|[-]+)[a-z0-9]+)*(?:\/[a-z0-9]+(?:(?:[._]|__|[-]+)[a-z0-9]+)*)*)(?::([A-Za-z0-9][A-Za-z0-9.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][0-9a-fA-F]{32,}))?/;
+
 function extractImages(
   content: string,
   config: ExtractConfig,
 ): PackageDependency[] {
   const deps: PackageDependency[] = [];
 
   for (const line of content.split(newlineRegex)) {
-    const match = regEx(/^\s*-?\s*image:\s*['"]?([^\s'"]+)['"]?\s*/).exec(line);
+    const pattern = new RegExp(
```

Comment:
why this triple regex object creation?

Reply:
I reduces to a single RegEx initialization with [`b4eeddf` (#31618)](https://github.com/renovatebot/renovate/pull/31618/commits/b4eeddf4ad07722cc78980246f80fe61473542a2).